### PR TITLE
jwt support was added

### PR DESCRIPTION
**Pull request by MaxSkory**

### CODE REVIEW

**Comment by MaxSkory** [comment]
Зробив окремий репо, тому, що це новий розділ, а я вже пройшов spring security.
https://github.com/MaxSkory/online-book-store

File: src/main/java/mskory/base/authentication/exception/JwtParsingException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package mskory.base.authentication.exception;
+
+public class JwtParsingException extends RuntimeException {
+    public JwtParsingException(String message, Throwable cause) {
+        super(message, cause);
+    }
+}
```
Comment (Sanderfuck) [inline src/main/java/mskory/base/authentication/exception/JwtParsingException.java]: Do you need 3 custom exception? You can use different messages for one custom

File: src/main/java/mskory/base/authentication/security/CustomUserDetailService.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```
Comment (Sanderfuck) [inline src/main/java/mskory/base/authentication/security/CustomUserDetailService.java]: Why you throws?

File: src/main/java/mskory/base/authentication/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String authHeader = request.getHeader(AUTHORIZATION_HEADER_NAME);
+        if (StringUtils.hasText(authHeader) && authHeader.startsWith(TOKEN_PREFIX)) {
+            return authHeader.substring(TOKEN_PREFIX.length());
+        }
+        return null;
```
Comment (Sanderfuck) [inline src/main/java/mskory/base/authentication/security/JwtAuthenticationFilter.java]: Return null is bad approach

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -8,5 +8,9 @@ spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
 spring.jackson.deserialization.fail-on-unknown-properties=true
 
+jwt.token.expiration-time=30
```
Comment (Sanderfuck) [inline src/main/resources/application.properties]: Is it enough?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 4 above)
Comment (MaxSkory) [reply src/main/resources/application.properties]: 30 minutes is not enough?

Comment (okuzan) [reply src/main/resources/application.properties]: I believe it's 30 seconds

Comment (MaxSkory) [reply src/main/resources/application.properties]: Знімок екрана 2023-11-20 о 15 38 34

**Pull request merged**